src/Console/Command/SelfUpdateCommand.php

```diff
@@ -0,0 +1,198 @@
+    const PACKAGIST_PACKAGE_NAME = 'humbug/php-scoper';
+    /** @internal */
+    const ROLLBACK_OPT = 'rollback';
+    /** @internal */
+    const CHECK_OPT = 'check';
+
+    /**
+     * @var Updater
+     */
+    private $updater;
+
+    /**
+     * @var OutputInterface
+     */
+    private $output;
+
+    /**
+     * @var string
+     */
+    private $version;
+
+    /**
+     * @var UpdateConsoleLogger
+     */
+    private $logger;
+
+    /**
+     * @param Updater $updater
+     */
+    public function __construct(Updater $updater)
+    {
+        parent::__construct();
+
+        $this->version = $this->getApplication()->getVersion();
+        $this->updater = $updater;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('self-update')
+            ->setDescription(sprintf(
+                    'Update %s to most recent stable build.',
+                    $this->getLocalPharName()
+            ))
+            ->addOption(
+                self::ROLLBACK_OPT,
+                'r',
+                InputOption::VALUE_NONE,
+                'Rollback to previous version of PHP-Scoper if available on filesystem.'
+            )
+            ->addOption(
+                self::CHECK_OPT,
+                'c',
+                InputOption::VALUE_NONE,
+                'Checks whether an update is available.'
+            )
+        ;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->logger = new UpdateConsoleLogger(
+            new SymfonyStyle($input, $output)
+        );
+
+        $this->output = $output;
+
+        if ($input->getOption('rollback')) {
+            $this->rollback();
+
+            return;
+        }
+
+        if ($input->getOption('check')) {
+            $this->printAvailableUpdates();
+
+            return;
+        }
+
+        $this->update($this->getStableUpdater());
+    }
+
+    /**
+     * @return Updater
+     */
+    private function getStableUpdater(): Updater
+    {
+        $this->updater->setStrategy(Updater::STRATEGY_GITHUB);
+        $this->updater->getStrategy()->setPackageName(self::PACKAGIST_PACKAGE_NAME);
+        $this->updater->getStrategy()->setPharName(self::REMOTE_FILENAME);
+        $this->updater->getStrategy()->setCurrentLocalVersion($this->version);
+
+        return $this->updater;
+    }
+
+    private function update(Updater $updater)
+    {
+        $this->logger->startUpdating();
+        try {
+            $result = $this->updater->update();
+
+            $newVersion = $this->updater->getNewVersion();
+            $oldVersion = $this->updater->getOldVersion();
+
+            if ($result) {
+                $this->logger->updateSuccess($newVersion, $oldVersion);
+            } else {
+                $this->logger->updateNotNeeded($oldVersion);
+            }
+        } catch (\Exception $e) {
```

Comment:
What kind of error can we get? If it's a generic one maybe we should use `Throwable` instead?